src/d3a/models/market/balancing.py

```diff
@@ -84,14 +90,22 @@ def accept_offer(self, offer_or_id: Union[str, BalancingOffer], buyer: str, *,
             if time is None:
                 time = self._now
             if energy is not None:
+
+                # reducing trade_rate to be charged in terms of grid_fee
+                if iaa_fee:
+                    source_rate = trade_rate / (1 + self.transfer_fee_ratio)
```

Comment:
Attention! a / (1+b) != a - a*b
I guess you were looking for source_rate = trade_rate * (1 - self.transfer_fee_ratio), right?

Reply:
No, its the other way around.
`target_rate/trade_rate = source_rate * (1 + self.transfer_fee_ratio)`.
I have rearranged it to get the source_rate.

Reply:
Ahh, I see. Sorry. Wanted to prevent you from a mistake that I was doing once. Thanks for explaining.